From 5e003b658d8edc7e08fd17b9f4aba8cd38788d33 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 24 Jul 2013 09:39:34 +0000
Subject: [PATCH 0368/1000] Mantis 24145: "Remove CORBA.SERVANTS configuration"
 - mStarter version 1.1.1.7 - Support $SERVANTPORT_LIST macro - Scan now all
 items in PLATFORM.EXEC configuration for $SERVANTPORT macros (previously just
 the items to execute).

git-svn-id: svn://localhost/SelfServiceCommon/trunk@542 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Controller/inc/mstarter_def.h      | 13 ++++++++++---
 .../Massai/cpp/Controller/src/mstarter_config.cpp | 15 +++++++++++++++
 .../Massai/cpp/Controller/src/serviceMap.cpp      |  2 +-
 .../cpp/Controller/src/setCORBASERVANTS.cpp       |  5 +++--
 4 files changed, 29 insertions(+), 6 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h b/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
index 15924703..4c69d3cb 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
+++ b/SelfServiceCommon/Massai/cpp/Controller/inc/mstarter_def.h
@@ -9,16 +9,23 @@
  *
  ************************************************************************/
 
-#define FILE_VER_BIN     1,1,1,6
-#define FILE_VER_STR     "FileVersion", "1.1.1.6\0"
+#define FILE_VER_BIN     1,1,1,7
+#define FILE_VER_STR     "FileVersion", "1.1.1.7\0"
 
 /*!
 \page massaistarter mStarter - Release Notes
 
-@version 1.1.1.6
+@version 1.1.1.7
 
 \section change1 Change history
 
+\date 24.07.2013 \li V1.1.1.7
+                 \li extension - JK Mantis 24145 Remove CORBA.SERVANTS configuration
+                 \li - activate SUPPORT_SERVANTPORT_LIST
+                 \li - scan not only the items configured foor execution in
+                 \li   PLATFORM.EXEC, but all entries there. The $SERVANTPORT_LIST will
+                 \li   contain $SERVANTPORT mentioned there too.
+
 \date 04.07.2013
 \li V1.1.1.6
 \li Mantis 0027896: mstarter: optimize start order
diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
index c547b4e7..d39e142e 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/mstarter_config.cpp
@@ -5,6 +5,7 @@
 #include "cfgclnt/nConfigSelection.hpp"
 #include "cfgclnt/nConfigNumber.hpp"
 #include "cfgclnt/nConfigTextList.hpp"
+#include "cfgclnt/nConfigBasicDirectory.hpp"
 #include "cfgclnt/nConfigBasicTextList.hpp"
 #include "boost/shared_ptr.hpp"
 #include "systools/mAutoMutexCS.h"
@@ -459,6 +460,20 @@ namespace mStarter
 
   PropertyList Config::getProcessList() const
   {
+    //
+    // Scan all EXEC configuration and substitute port macros there
+    std::string dir_path( execSection );
+    nConfig::basicDirectory cfg_dir( "mstarter" );
+    std::list<std::string> const cfg_dir_items = cfg_dir.getItems( dir_path.substr( 0, dir_path.size() - 1 ).c_str() );
+    for(
+         std::list<std::string>::const_iterator dir_it =
+            cfg_dir_items.begin();
+         dir_it != cfg_dir_items.end();
+         ++dir_it
+       )
+    {
+       m_substituteEnv.substEnvList( m_execList->getValue( dir_it->c_str() ) );
+    }
     TextList components;
     m_softwareList->appendTo(components);
     m_hardwareList->appendTo(components);
diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/serviceMap.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/serviceMap.cpp
index 4871aadf..23a13bef 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/serviceMap.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/serviceMap.cpp
@@ -16,7 +16,7 @@
 // #define SETUP_CORBA_SERVANTS
 
 //! Feature configuration: Do we implement the SERVANTPORT_LIST macro?
-// #define SUPPORT_SERVANTPORT_LIST
+#define SUPPORT_SERVANTPORT_LIST
 
 #if defined(_MSC_VER)
 // That compiler cannot handle names in debug information > 255 character.
diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/setCORBASERVANTS.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/setCORBASERVANTS.cpp
index 8c4266df..66b33d73 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/setCORBASERVANTS.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/setCORBASERVANTS.cpp
@@ -12,7 +12,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2012 MATERNA Information & Communications
+ * Copyright (c) 2012-2013 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -87,7 +87,8 @@ int main( int argc, char ** argv )
       //
       port_servant = args.argument( ++argx );
    }
-  closeTrace();
+   closeTrace();
+   return 0;
 }
 
 
-- 
2.41.0.windows.1

